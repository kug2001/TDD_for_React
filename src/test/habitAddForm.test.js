import React from 'react';
import { screen, render } from '@testing-library/react';
import HabitAddForm from '../components/habitAddForm';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

describe('AddForm 기능 테스트 코드', () => {
  it('렌더 스냅샷 테스트', () => {
    const componet = renderer.create(<HabitAddForm onAdd={jest.fn()} />);
    expect(componet.toJSON()).toMatchSnapshot();
  })
  describe('add함수 테스트', () => {
    let onAdd;
    let input;
    let button;
    beforeEach(() => {
      onAdd = jest.fn();
      render(<HabitAddForm onAdd={onAdd} />);
      input = screen.getByPlaceholderText('Habit');
      button = screen.getByText('Add');
    })
    it('add버튼 클릭시 onAdd()에 input value를 담아서 호출', () => {
      // input에다가 원하는 습관의 이름을 타이핑 한 다음에
      userEvent.type(input, 'sleep');
      // add라는 버튼을 클릭하면
      userEvent.click(button);
      // onAdd가 input에 입력된 이름과 함께 호출되어야 함.
      expect(onAdd).toHaveBeenCalledWith('sleep');
      expect(onAdd).toHaveBeenCalledTimes(1);
    })
    it('input데이터가 없을 때 add버튼 클릭시 onAdd() 호출 안함', () => {
      // input에다가 원하는 습관의 타이핑이 없게 한다음에
      userEvent.type(input, '');
      // add라는 버튼을 클릭하면
      userEvent.click(button);
      // onAdd 호출되면 안됨
      expect(onAdd).toHaveBeenCalledTimes(0);
    })
  })

})