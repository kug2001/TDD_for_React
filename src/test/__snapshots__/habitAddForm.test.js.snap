// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm 기능 테스트 코드 렌더 스냅샷 테스트 1`] = `
<form
  className="add-form"
  onSubmit={[Function]}
>
  <input
    className="add-input"
    placeholder="Habit"
    type="text"
  />
  <button
    className="add-button"
  >
    Add
  </button>
</form>
`;
